### PR TITLE
feat(core): add nx task planner

### PR DESCRIPTION
**Pull request by Cammisuli**
## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/nx/src/native/tasks/dep_outputs.rs [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+    for task_dep in &task_graph.dependencies[task.id.as_str()] {
+        let child_task = &task_graph.tasks[task_dep.as_str()];
+
+        let now = std::time::Instant::now();
+        let outputs = get_outputs_for_target_and_configuration(
+            child_task,
+            &project_graph.nodes[&child_task.target.project],
+        )?;
+        let output_files = get_files_for_outputs(workspace_root.to_string(), outputs)?;
+        println!(
```
Comment (FrozenPandaz) [inline packages/nx/src/native/tasks/dep_outputs.rs:36]: Change to debugLog

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: packages/nx/src/native/tasks/hash_planner.rs [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+            workspace_root,
+        }
+    }
+
+    #[napi]
+    pub fn get_plans(
+        &self,
+        task_ids: Vec<&str>,
+        task_graph: TaskGraph,
+    ) -> anyhow::Result<HashMap<String, Vec<String>>> {
```
Comment (FrozenPandaz) [inline packages/nx/src/native/tasks/hash_planner.rs:39]: Create a separate function which keeps it as a `HashInstruction` which will be easier to work with within Rust.

Comment (FrozenPandaz) [reply packages/nx/src/native/tasks/hash_planner.rs:39]: enum HashInstruction {
  'AllExternalDependencies'
  'ExternalNode'
  'FileSet'
  'EnvironmentVariable'
  'Runtime'
  'ProjectConfiguration'
  'TsConfiguration'
  'TaskOutput'
}

File: packages/nx/src/native/tasks/inputs.rs [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+    task: &'a Task,
+    project_graph: &'a ProjectGraph,
+    nx_json: &'a NxJson,
+) -> anyhow::Result<SplitInputs<'a>> {
+    let project_node = project_graph
+        .nodes
+        .get(&task.target.project)
+        .expect("Task target should always have a project");
+    let named_inputs = get_named_inputs(nx_json, project_node);
+    let target_data = project_node
```
[the remark this answers is not in the document]
Comment (Cammisuli) [reply packages/nx/src/native/tasks/inputs.rs:24]: Tests were failing without the `targetDefaults`. I migrated most of the tests from `task-hasher.spec.ts` which specifically has a test called `"should use targetDefaults from nx.json"`

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**